src/emulator/auth/operations.ts

```diff
@@ -223,12 +227,39 @@ function signUp(
     ({ user } = parseIdToken(state, reqBody.idToken));
   }
 
+  let extraClaims;
   if (!user) {
-    if (reqBody.localId) {
-      user = state.createUserWithLocalId(reqBody.localId, updates);
-      assert(user, "DUPLICATE_LOCAL_ID");
-    } else {
-      user = state.createUser(updates);
+    updates.createdAt = timestamp.getTime().toString();
+    const localId = reqBody.localId ?? state.generateLocalId();
+    if (reqBody.email) {
+      const userBeforeCreate = { localId, ...updates };
+      const blockingResponse = await fetchBlockingFunction(
+        state,
+        BlockingFunctionEvents.BEFORE_CREATE,
+        userBeforeCreate,
+        { signInMethod: "password" }
+      );
+      updates = { ...updates, ...blockingResponse.updates };
+    }
+
+    user = state.createUserWithLocalId(localId, updates);
+    assert(user, "DUPLICATE_LOCAL_ID");
+
+    if (reqBody.email) {
+      if (!user.disabled) {
+        const blockingResponse = await fetchBlockingFunction(
```

Comment:
I believe the current logic calls the `beforeSignIn` blocking function even for an admin request (e.g. from AdQmin SDKs or Emulator UI). Is this the right behavior?